plugins/git-build.js

```diff
@@ -37,7 +37,10 @@ function buildConsumer(config, cimpler, repoPath) {
          timeout: config.timeout || 0,
          maxBuffer: config.maxBuffer || 1024 * 1024 * 2
       },
-      killChildrenOnExit = "trap '[ $(jobs -p) ] && kill $(jobs -p)' EXIT",
+      killChildrenOnExit = `trap '[ -z "$(jobs -p)" ] || kill -SIGTERM $(jobs -p) \
+         && sleep 2 \ 
+         && [ -z "$(jobs -p)" ] || kill -SIGKILL $(jobs -p); \
+         wait $(jobs -p)' EXIT`,
```

Comment:
Whew. This is a doozy. I had zero chance of verifying this without playing around in bash. Having done that, I have a couple comments which aren't strong enough to be suggestions, let alone requests:

- The backslashes aren't necessary (but they don't hurt). We're inside a single-quoted string, and bash is going to read the whole thing and treat it as an expression. Newlines in the string won't be replaced with semicolons.
- I don't love the mixing of `||` and `&&` without explicit grouping, but it works fine because they're of equal precedence in bash. Also, the only option for explicit grouping without a subshell would look like `{ foo || bar; }`, where the semicolon is mandatory, and I don't love that either.
- `wait` without arguments will, by default, wait on all active child processes.
- I found out that the trap expression is only going to run on an exit of 0. That isn't a new thing, but something to be aware of. If the job is killed or exits with a non-zero code, none of this happens.

Because this is a complicated expression, I propose restructuring just a little bit to help highlight the different components:

```js
killChildrenOnExit = `
   trap '
      [ -z "$(jobs -p)" ] || kill -SIGTERM $(jobs -p)
         && sleep 2
         && [ -z "$(jobs -p)" ] || kill -SIGKILL $(jobs -p)
         ; wait $(jobs -p)
      '
      EXIT`,
```